src/daemon/database/actions.rs

```diff
@@ -1708,4 +1780,110 @@ mod test {
+        )
+        .unwrap();
+        let amount = Amount::from_sat(22400000);
+        let derivation_index = ChildNumber::from(1);
+        db_insert_new_unconfirmed_vault(
+            &db_path,
+            wallet_id,
+            &outpoint_b,
+            &amount,
+            derivation_index,
+        )
+        .unwrap();
+        let db_vault = db_vault_by_deposit(&db_path, &outpoint_b).unwrap().unwrap();
+        db_confirm_deposit(
+            &db_path,
+            &outpoint_b,
+            9,
+            9,
+            &fresh_unvault_tx,
+            &fresh_cancel_tx,
+            Some(&fresh_emergency_tx),
+            Some(&fresh_unvaultemergency_tx),
+        )
+        .unwrap();
+
+        let stored_cancel_tx = db_cancel_transaction(&db_path, db_vault.id)
+            .unwrap()
+            .unwrap();
+        update_presigned_tx(
+            &db_path,
+            &db_vault,
+            stored_cancel_tx,
+            &fullysigned_cancel_tx.psbt().inputs[0].partial_sigs,
+            &revaultd.secp_ctx,
+        );
+
+        let stored_emergency_tx = db_emer_transaction(&db_path, db_vault.id).unwrap().unwrap();
+        update_presigned_tx(
+            &db_path,
+            &db_vault,
+            stored_emergency_tx,
+            &fullysigned_emergency_tx.psbt().inputs[0].partial_sigs,
+            &revaultd.secp_ctx,
+        );
+
+        let stored_unvaultemergency_tx = db_unvault_emer_transaction(&db_path, db_vault.id)
+            .unwrap()
+            .unwrap();
+        update_presigned_tx(
+            &db_path,
+            &db_vault,
+            stored_unvaultemergency_tx,
+            &fullysigned_unvaultemergency_tx.psbt().inputs[0].partial_sigs,
+            &revaultd.secp_ctx,
+        );
+
+        db_update_vault_status(&db_path, &db_vault).unwrap();
+        let db_vault = db_vault_by_deposit(&db_path, &db_vault.deposit_outpoint)
+            .unwrap()
+            .unwrap();
+        assert_eq!(db_vault.status, VaultStatus::Secured);
+        assert!(db_vault.funded_at.is_some());
+        assert!(db_vault.secured_at.is_some());
+        assert!(db_vault.delegated_at.is_none());
+
+        let stored_unvault_tx = db_unvault_transaction(&db_path, db_vault.id)
+            .unwrap()
+            .unwrap();
+        update_presigned_tx(
+            &db_path,
+            &db_vault,
+            stored_unvault_tx,
+            &fullysigned_unvault_tx.psbt().inputs[0].partial_sigs,
+            &revaultd.secp_ctx,
+        );
+        db_update_vault_status(&db_path, &db_vault).unwrap();
+        let db_vault = db_vault_by_deposit(&db_path, &db_vault.deposit_outpoint)
+            .unwrap()
+            .unwrap();
+        assert_eq!(db_vault.status, VaultStatus::Active);
+        assert!(db_vault.funded_at.is_some());
+        assert!(db_vault.secured_at.is_some());
+        assert!(db_vault.delegated_at.is_some());
+    }
```

Comment:
Arf, overlooked this. You forgot to remove the datadir.

Reply:
My bad :facepalm:  https://github.com/revault/revaultd/pull/328